spec/spec_helper.rb

```diff
@@ -48,6 +48,7 @@ def serve_once(body)
     yield
   ensure
     server.shutdown
+    thread.join(5)
     thread.exit
```

Comment:
No the `5` makes it return after five seconds regardless. I observed a race condition with `thread.exit` causing this to quit so fast that the next `serve_once` would find to bind to the port since it hasn't been freed yet, since the shutdown wasn't completed yet I guess. However on 2.1 `#join` would never return for some reason, this seems to work around both conditions so far.